Gordon360/ApiControllers/ScheduleControlController.cs

```diff
@@ -0,0 +1,196 @@
+        public IHttpActionResult Get(string username)
+        {
+            object scheduleControlResult = null;
+
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+            scheduleControlResult = _unitOfWork.ScheduleControlRepository.GetById(id);
+            
+            if (scheduleControlResult == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(scheduleControlResult);
+
+        }
+
+
+
+        /// <summary>
+        /// Update privacy of schedule
+        /// </summary>
+        /// <param name="value">Y or N</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("update/privacy/{value}")]
+        public IHttpActionResult UpdateSchedulePrivacy(string value)
+        {
+            // Verify Input
+            if (!ModelState.IsValid)
+            {
+                string errors = "";
+                foreach (var modelstate in ModelState.Values)
+                {
+                    foreach (var error in modelstate.Errors)
+                    {
+                        errors += "|" + error.ErrorMessage + "|" + error.Exception;
+                    }
+
+                }
+                throw new BadInputException() { ExceptionMessage = errors };
+            }
+
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var id = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "id").Value;
+            _scheduleControlService.UpdateSchedulePrivacy(id, value);
+
+            return Ok();
+
+        }
+
+        /// <summary>
+        /// Update schedule description
+        /// </summary>
+        /// <param name="value">New description</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("update/description/{value}")]
+        public IHttpActionResult UpdateDescription(string value)
+        {
+            // Verify Input
+            if (!ModelState.IsValid)
+            {
+                string errors = "";
+                foreach (var modelstate in ModelState.Values)
+                {
+                    foreach (var error in modelstate.Errors)
+                    {
+                        errors += "|" + error.ErrorMessage + "|" + error.Exception;
+                    }
+
+                }
+                throw new BadInputException() { ExceptionMessage = errors };
+            }
+
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var id = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "id").Value;
+            _scheduleControlService.UpdateDescription(id, value);
+
+            return Ok();
+
+        }
+
+        /// <summary>
+        /// Update timestamp of last modified schedule
+        /// </summary>
+        /// <param name="value">Datetime in string</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("update/timestamp/{value}")]
```

Comment:
Do we need update and the value in the URL?

Reply:
[https://github.com/gordon-cs/gordon-360-api/pull/252#discussion_r303606771](url)